src/solver/optim/ortoolsImpl/linearProblem.cpp

```diff
@@ -0,0 +1,215 @@
+{
+    return addVariable(lb, ub, true, name);
+}
+
+OrtoolsMipVariable* OrtoolsLinearProblem::getVariable(const std::string& name) const
+{
+    return variables_.at(name).get();
+}
+
+OrtoolsMipConstraint* OrtoolsLinearProblem::addConstraint(double lb,
+                                                          double ub,
+                                                          const std::string& name)
+{
+    if (constraints_.contains(name))
+    {
+        logs.error() << "This constraint already exists: " << name;
+        throw ElemAlreadyExists();
+    }
+
+    auto* mpConstraint = mpSolver_->MakeRowConstraint(lb, ub, name);
+    auto mipConstraint = std::make_unique<OrtoolsMipConstraint>(mpConstraint);
+
+    if (!mpConstraint || !mipConstraint)
+    {
+        logs.error() << "Couldn't add variable to Ortools MPSolver: " << name;
+    }
+
+    const auto& mapIteratorPair = constraints_.try_emplace(name, std::move(mipConstraint));
+
+    if (!mapIteratorPair.second)
+    {
+        logs.error() << "Error adding variable: " << name;
+    }
+
+    return mapIteratorPair.first->second.get(); // <<name, constraint>, bool>
+}
+
+OrtoolsMipConstraint* OrtoolsLinearProblem::getConstraint(const std::string& name) const
+{
+    return constraints_.at(name).get();
+}
+
+static const operations_research::MPVariable* getMpVar(const Api::IMipVariable* var)
+
+{
+    const auto* orMpVar = dynamic_cast<const OrtoolsMipVariable*>(var);
+    if (!orMpVar)
+    {
+        logs.error() << "Invalid cast, tried from Api::IMipVariable to OrtoolsMipVariable";
+        throw std::bad_cast();
+    }
+    return orMpVar->getMpVar();
+}
+
+void OrtoolsLinearProblem::setObjectiveCoefficient(Api::IMipVariable* var, double coefficient)
+{
+    objective_->SetCoefficient(getMpVar(var), coefficient);
+}
+
+double OrtoolsLinearProblem::getObjectiveCoefficient(const Api::IMipVariable* var) const
+{
+    return objective_->GetCoefficient(getMpVar(var));
+}
+
+void OrtoolsLinearProblem::setMinimization()
+{
+    objective_->SetMinimization();
+}
+
+void OrtoolsLinearProblem::setMaximization()
+{
+    objective_->SetMaximization();
+}
+
+bool OrtoolsLinearProblem::isMinimization() const
+{
+    return objective_->minimization();
+}
+
+bool OrtoolsLinearProblem::isMaximization() const
+{
+    return objective_->maximization();
+}
+
+static Api::MipStatus convertStatus(operations_research::MPSolver::ResultStatus& status)
+{
+    switch (status)
+    {
+    case operations_research::MPSolver::ResultStatus::OPTIMAL:
+        return Api::MipStatus::OPTIMAL;
+    case operations_research::MPSolver::ResultStatus::FEASIBLE:
+        return Api::MipStatus::FEASIBLE;
+    case operations_research::MPSolver::ResultStatus::UNBOUNDED:
+        return Api::MipStatus::UNBOUNDED;
+    default:
+        logs.warning() << "Solve returned an error status";
+        break;
+    }
+    return Api::MipStatus::MIP_ERROR;
+}
+
+OrtoolsMipSolution* OrtoolsLinearProblem::solve(bool verboseSolver)
```

Comment:
In the same spirit as for [this comment](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2286#discussion_r1741760630) (in the associated **.h** header), this **solve(...)** method could be simpler using **ortools** native types as most as possible : 
```c++
OrtoolsMipSolution* OrtoolsLinearProblem::solve(bool verboseSolver)
{
    if (verboseSolver)
    {
        mpSolver_->EnableOutput();
    }

    auto mpStatus = mpSolver_->Solve(params_);
    return std::make_unique<OrtoolsMipSolution>(mpSolver_).get();
}
```
This function is now simple. 
Indeed, some task are now delegated to class **OrtoolsMipSolution** : it's more the mission of class **OrtoolsMipSolution** to deal with **solution** details, not the job of **OrtoolsLinearProblem**

Reply:
```cpp
    return std::make_unique<OrtoolsMipSolution>(mpSolver_).get();
```
dangling pointer 👎🏻

Reply:
```cpp
    return std::make_unique<OrtoolsMipSolution>(mpSolver_).get();
```
dangling pointer 👎🏻

Reply:
So I guess a solution would be to return a smart pointer **OrtoolsMipSolution**, don't you think ?

Reply:
We could return a `std::shared_ptr` but it doesn't help much.

Reply:
> We could return a `std::shared_ptr` but it doesn't help much.

Where is the problem with smart pointers ?

Reply:
Performance & cycles

Reply:
**Summary** : 
We are currently forced to keep 2 lists of (std::unique_ptr) pointers : 
- **variables_** : pointer to **OrtoolsMipVariable**
- **constraints_** : pointers to **OrtoolsMipConstraint**

These lists are required for member functions `addVariable(...)`, `getVariable(...,)` to return a **raw** pointer to **OrtoolsMipVariable** with respect to covariance related to base class **ILinearProblem** (same for constraints).

An subject could be to simplify code by removing these lists if we think they're not required.
To do that, instead of a raw pointer, these functions could return a freshly created **std** smart pointer.
A problem could be about performances, hat we need to test.
We should create a ticket for this.

Reply:
See #2367

Reply:
See also [this ticket](https://github.com/AntaresSimulatorTeam/Antares_Simulator/issues/2386)